File: ruby/lib/MessageBroker/dispatchers/RabbitMQ/routings/BaseRabbitMQRoom.rb
require_relative '../../IRoom'

class BaseRabbitMQRoom
  def initialize queue:
    @queue = queue
  end

  def subscribe block: true
    begin
      @queue.subscribe(block: block) do |delivery_info, properties, payload|
        yield delivery_info, properties, payload
      end
    rescue Interrupt => _
    
    end
  end

  implements IRoom
end

File: js/lib/MessageBroker/dispatchers/RabbitMQ/routings/BaseRabbitMQRoom.js
const IRoom = require("../../IRoom");

class BaseRabbitMQRoom extends IRoom {
  constructor({ channel, queue }) {
    super();

    this.channel = channel;
    this.queue = queue;
  }

  subscribe(callback) {
    this.channel.consume(this.queue, callback, {noAck: true});
  }
}

module.exports = BaseRabbitMQRoom